src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -1653,6 +1653,13 @@ List<JCExpression> loadFreevars(DiagnosticPosition pos, List<VarSymbol> freevars
     }
 //where
         JCExpression loadFreevar(DiagnosticPosition pos, VarSymbol v) {
+
+            // Verify we are not trying to access a proxy field from a static method
+            Symbol proxy = proxies.get(v);
+            if (proxy != null && proxy.owner == currentClass && currentMethodSym.isStatic())
+                log.error(pos, Errors.NonStaticCantBeRef(Kinds.kindName(v), v));
```

Comment:
we should never issue errors in Lower (unless for sanity checks). Note that most IDEs runs javac in attribution-only mode, so these errors are not reported. It seems to be that a check is missing here, and this is actually related to the comment I made here:

https://github.com/openjdk/jdk/pull/19705#issuecomment-2174411126

Reply:
OK thanks. I didn't know about the IDE thing - makes sense. In any case I couldn't figure out how to do it in `Attr.java` and so resorted to this.